kairon/shared/data/customer_order_processor.py

```
@@ -0,0 +1,222 @@
+                persona_type=persona_type,
+                user=user,
+                **{k: v for k, v in payload.items() if k not in ("user", "bot", "user_id")},
+            )
+
+
+        customer.address_list = [Address(**a) for a in address_list_raw]
+
+        try:
+            customer.save()
+        except NotUniqueError:
+            raise AppException("Customer with this identifier already exists for another bot")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)